src/expression/mod.rs

```diff
@@ -141,11 +141,11 @@ impl<F: PrimeField> Expression<F> {
     }
 
     pub fn scalar_product(a: Vec<Expression<F>>, b: Vec<Expression<F>>) -> Expression<F> {
-        a.into_iter().zip(b.into_iter()).map(|(a, b)| a * b).sum()
```

Comment:
Cool! didn't know you could do omit the `.into_iter()` call here

Reply:
`zip` just requires the arguments to implement `IntoIterator`: https://doc.rust-lang.org/beta/std/iter/trait.IntoIterator.html

which [`Vec` does out of the box](https://doc.rust-lang.org/beta/std/iter/trait.IntoIterator.html#implementors)